src/main/api/fileStore.ts

```diff
@@ -0,0 +1,79 @@
+import path from 'path'
+
+import { app, ipcRenderer, ipcMain } from 'electron'
+import * as fs from 'fs-extra'
+
+import { ApiFileStoreService, StoreFileData, StoreFileName } from '../../shared/api/types'
+import { getStoreFilesIPCMessages } from '../../shared/ipc/fileStore'
+
+const APP_NAME = app?.name ?? 'ASGARDEX'
+
+const APP_DATA_DIR = path.join(app?.getPath('appData') ?? './testdata', APP_NAME)
+export const STORAGE_DIR = path.join(APP_DATA_DIR, 'storage')
+
+// If `fullFilePathname' does not exist, `fs.remove` silently does nothing.
+// @see https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md
+const removeFile = async (fullFilePathname: string) => fs.remove(fullFilePathname)
+
+const isFileExist = async (fullFilePathname: string) => fs.pathExists(fullFilePathname)
+
+const getFileContent = async <T>(fullFilePathname: string, defaultValue: T): Promise<T> => {
+  // If wile does not exist create a new one with defaultValue as content
+  if (!(await isFileExist(fullFilePathname))) {
+    await fs.writeJSON(fullFilePathname, { ...defaultValue })
+    return Promise.resolve(defaultValue)
+  }
+
+  try {
+    // Try to read and parse appropriate JSON store-file
+    const fileContent = await fs.readJSON(fullFilePathname)
+    // Combine file content with provided default value
+    return Promise.resolve({ ...defaultValue, ...fileContent })
+  } catch {
+    // Resolve with default value in case of any error
+    return Promise.resolve(defaultValue)
+  }
+}
+/**
+ * Replaces store-file's content with partial provided data
+ */
+const saveFile = async <T>(fullFilePathname: string, data: Partial<T>, defaultValue: T) => {
```

Comment:
`saveFile` might be misleading here, since it does not just save some data into a file, it append data to existing stuff. Maybe `appendDataToFile` or something ....?

Reply:
hm. actually `appendDataToFile` is misleading too i believe. in my understanding append will work like:
old file content
`{"some": "data here}`
after append {newData: "new data here"} i assume content to be smth like this
`{"some": "data here}{"newData: "new data here""}`
as `append` method of the `fs` package

Reply:
`saveToFile` :100: